src/message/handlers/download.ts

```diff
@@ -21,11 +21,11 @@ const cancelDownloadTask = () => {
 export const messageHandlers: Partial<MessageHandlers> = {
   [MessageType.DOWNLOAD_TRACK]: async ({ message }) => {
     const fileUrl = message.urls.find(url => url !== null && url !== undefined)
-    const fileExtension = fileUrl.split('.').pop()
-    const fileName = message.title + '.' + fileExtension
+    const fileName = message.filename
+    const trackName = fileName.split('.').slice(0, -1).join('')
 
     dispatch(setVisibility({ drawer: 'DownloadTrackProgress', visible: true }))
-    dispatch(setFileInfo({ trackName: message.title, fileName: fileName }))
+    dispatch(setFileInfo({ trackName: trackName, fileName: fileName }))
```

Comment:
Same nit as from the other PR
```suggestion
    dispatch(setFileInfo({ trackName, fileName }))
```